materialsLibrary/src/mix/mixMaterial.ts

```diff
@@ -171,20 +171,48 @@ export class MixMaterial extends PushMaterial {
 
         // Textures
         if (scene.texturesEnabled) {
+            if (!this._mixTexture1 || !this._mixTexture1.isReady()) {
+                return false;
+            }
+
+            defines._needUVs = true;
+
+            if (this._mixTexture2) {
+                defines.MIXMAP2 = true;
+            }
+            if (!this._mixTexture2 || !this._mixTexture2.isReady()) {

```

Comment:
My bad, I think this test should move in line 205
If (mixTexture2) { ... }
To test that if mixTexture2 then check if mixmap2 is ready and all other diffuse textures for mixmap2. You can also add the define in this test